src/credentials.c

```diff
@@ -57,19 +49,23 @@ void aws_crt_credentials_options_set_expiration_timepoint_seconds(
 
 aws_crt_credentials *aws_crt_credentials_new(const aws_crt_credentials_options *options) {
     struct aws_allocator *allocator = aws_crt_default_allocator();
-    return aws_credentials_new(
+    aws_crt_credentials *creds = aws_crt_resource_new(aws_crt_mem_calloc(1, sizeof(aws_crt_credentials)));
+    creds->credentials = aws_credentials_new(
         allocator,
         aws_byte_cursor_from_buf(&options->access_key_id),
         aws_byte_cursor_from_buf(&options->secret_access_key),
         aws_byte_cursor_from_buf(&options->session_token),
         options->expiration_timepoint_seconds);
+    return creds;
 }
 
 aws_crt_credentials *aws_crt_credentials_acquire(aws_crt_credentials *credentials) {
-    aws_credentials_acquire(credentials);
+    aws_credentials_acquire(credentials->credentials);
+    aws_crt_resource_acquire(&credentials->resource);
```

Comment:
This is kind of awkward.  Wondering if letting the crt resource track as a void * the actual object and take the release function as an additional parameter in the creation function.

```
struct aws_crt_resource {
  struct aws_ref_count ref_count;
  void *resource;
  void *user_data;
  void blah blah user data destructor_fn;
  void blah blah resource_release_fn;
};
```
Unsure if this is an improvement, but double refing feels weirrrrd.

Reply:
so the aws_crt_resource actually DOES track the object, and is responsible for its death. I could add a dtor, but that doesn't solve the problem for acquire, the ref count won't go up. Maybe it shouldn't? Either way, this is something we can fix under the hood, it doesn't change the API.